autopush/router/gcm.py

```diff
@@ -43,18 +45,20 @@ def register(self, uaid, router_data, router_token=None, *kwargs):
         if "token" not in router_data:
             raise self._error("connect info missing GCM Instance 'token'",
                               status=401)
+        if not (router_token == router_data['token'] == self.senderID):
```

Comment:
I don't really understand why the router_token needs to match the self.senderID, maybe a link to some GCM docs or something highlighting why this needs to match here? Does this also imply that every single GCM client stores a token in our DB that matches a hard-coded value of self.senderID?